controllers/kubeadmbootstrapconfig_controller.go

```diff
@@ -36,17 +39,45 @@ type KubeadmBootstrapConfigReconciler struct {
 
 // +kubebuilder:rbac:groups=kubeadm.bootstrap.cluster.sigs.k8s.io,resources=kubeadmbootstrapconfigs,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=kubeadm.bootstrap.cluster.sigs.k8s.io,resources=kubeadmbootstrapconfigs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=cluster.sigs.k8s.io,resources=clusters;machines,verbs=get;list;watch
 
 // Reconcile TODO
 func (r *KubeadmBootstrapConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Log.WithValues("kubeadmbootstrapconfig", req.NamespacedName)
 
+	config := &kubeadmv1alpha1.KubeadmBootstrapConfig{}
+	if err := r.Get(ctx, req.NamespacedName, config); err != nil {
+		log.Error(err, "stacktrace", fmt.Sprintf("%+v", err))
+	}
+
+	// find the first machine reference
+	var machineRef v1.OwnerReference
+	for _, ref := range config.OwnerReferences {
+		if ref.Kind == "Machine" {
+			machineRef = ref
+			break
+		}
+	}
+
 	machine := &clusterapiv1alpha2.Machine{}
-	if err := r.Get(ctx, req.NamespacedName, machine); err != nil {
+	machineKey := types.NamespacedName{
+		Name:      machineRef.Name,
+		Namespace: config.Namespace, // must be in the same namespace
+	}
+	if err := r.Get(ctx, machineKey, machine); err != nil {
 		log.Error(err, "stacktrace", fmt.Sprintf("%+v", err))
 	}
 
+	cluster := &clusterapiv1alpha2.Cluster{}
+	clusterKey := types.NamespacedName{
+		Name:      machine.ClusterName,
+		Namespace: config.Namespace, // must still be in the same namespace
+	}
+	if err := r.Get(ctx, clusterKey, cluster); err != nil {
+		log.Error(err, "stacktrace", fmt.Sprintf("%+v", err))
+	}
+	log.Info("debug", "cluster", cluster, "machine", machine, "config", config)
```

Comment:
Extra?

Reply:
yeah, not necessary, will not show up in the following PRs